tools/build_langserver/langserver/analyzer.go

```diff
@@ -887,3 +924,25 @@ func aspListToStrSlice(listVal *asp.List) []string {
 	}
 	return retSlice
 }
+
+// getCoreBuildLabel returns a core.BuildLabel object providing a BuildDef and its URI
+func getCoreBuildLabel(def *BuildDef, uri lsp.DocumentURI) (buildLabel core.BuildLabel, err error) {
+	fp, err := GetPathFromURL(uri, "file")
+	if err != nil {
+		return core.BuildLabel{}, err
+	}
+
+	rel, err := filepath.Rel(core.RepoRoot, filepath.Dir(fp))
+	if err != nil {
+		return core.BuildLabel{}, err
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			log.Warning("error occurred parsing build label")
+			err = r.(error)
+		}
+	}()
+
+	return core.NewBuildLabel(rel, def.BuildDefName), err
```

Comment:
FYI I think there's a `TryNewBuildLabel` or smthn that returns an error instead of panicking, that'd save the whole recover thing.